node/src/proxy_server/mod.rs

```diff
@@ -835,13 +698,23 @@ impl ProxyServer {
                     &payload_destination_key,
                 )
                 .expect("Key magically disappeared");
-                ProxyServer::report_routing_service(
-                    accountant_routing_sub,
-                    expected_services,
-                    timestamp,
-                    pkg.payload.len(),
-                    logger,
-                );
+                if !expected_services
```

Comment:
I think we never use the Proxy Server at all if we're in Zero-Hop mode. If that's the case (maybe some checking would be in order), then we don't have to do checks like this.